plugins/CuraEngineBackend/ProcessSlicedLayersJob.py

```diff
@@ -144,7 +159,33 @@ def run(self):
                 self._progress.setProgress(progress)
 
         # We are done processing all the layers we got from the engine, now create a mesh out of the data
-        layer_mesh = layer_data.build()
+
+        # Find out colors per extruder
+        global_container_stack = Application.getInstance().getGlobalContainerStack()
+        manager = ExtruderManager.getInstance()
+        extruders = list(manager.getMachineExtruders(global_container_stack.getId()))
```

Comment:
Easier debugging and not sure if an iterator will evaluate to False if it doesn't return any objects.